query/sources/impl/src/main/java/org/codice/ddf/admin/sources/wfs/WfsSourceUtils.java

```diff
@@ -179,52 +175,26 @@ public WfsSourceUtils(ConfiguratorSuite configuratorSuite) {
                     .evaluate(capabilitiesXml);
         } catch (XPathExpressionException e) {
             LOGGER.debug("Failed to parse XML response.");
-            configResult.addArgumentMessage(unknownEndpointError(responseField.requestUrlField()
+            return Reports.from(unknownEndpointError(responseField.requestUrlField()
                     .path()));
-            return configResult;
         }
 
-        configResult.result(preferredConfig.wfsVersion(wfsVersion));
-        if (!preferredConfig.validate()
-                .isEmpty()) {
-            configResult.addArgumentMessage(unknownEndpointError(responseField.requestUrlField()
+        WfsVersion wfsVersionToCheck = new WfsVersion();
+        wfsVersionToCheck.isRequired(true);
+        wfsVersionToCheck.setValue(wfsVersion);
+        if(!wfsVersionToCheck.validate().isEmpty()) {
+            return Reports.from(unknownEndpointError(responseField.requestUrlField()
                     .path()));
         }
 
-        return configResult;
-    }
-
-    private class WfsTaskCallable
-            implements Callable<ReportWithResultImpl<WfsSourceConfigurationField>> {
-
-        private String url;
-
-        private HostField host;
-
-        private CredentialsField creds;
-
-        public WfsTaskCallable(String url, HostField host, CredentialsField creds) {
-            this.url = url;
-            this.host = host;
-            this.creds = creds;
-        }
-
-        @Override
-        public ReportWithResultImpl<WfsSourceConfigurationField> call() throws Exception {
-            UrlField requestUrl = new UrlField();
-            String formattedUrl = String.format(url, host.hostname(), host.port());
-            requestUrl.setValue(formattedUrl);
-
-            ReportWithResultImpl<WfsSourceConfigurationField> configResult = getWfsConfigFromUrl(
-                    requestUrl,
-                    creds);
-
-            if (!configResult.containsErrorMsgs()) {
-                return configResult;
-            }
+        WfsSourceConfigurationField wfsSourceConfigurationField = new WfsSourceConfigurationField()
+                .wfsVersion(wfsVersion);
+        wfsSourceConfigurationField.endpointUrl(requestUrl.getValue())
```

Comment:
I can't build and return in the same line because when we do `.username` and `.password` we get back a `CredentialField` so it says `Incompatible Types`